src/extension/background-script/actions/nostr/getPublicKeyOrPrompt.ts

```diff
@@ -9,26 +10,58 @@ const getPublicKeyOrPrompt = async (message: MessagePublicKeyGet) => {
     return;
   }
 
-  const result = await prompt(message);
+  const allowance = await db.allowances.get({
+    host: message.origin.host,
+  });
 
-  return result;
-};
+  if (!allowance?.id) {
+    return { error: "Could not find an allowance for this host" };
+  }
 
-const prompt = async (message: MessagePublicKeyGet) => {
-  try {
-    const response = await utils.openPrompt({
-      args: {},
-      ...message,
-      action: "public/nostr/confirmGetPublicKey",
-    });
+  const findPermission = await db.permissions.get({
+    host: message.origin.host,
+    method: PermissionMethod["NOSTR_GETPUBLICKEY"],
+  });
 
-    const publicKey = state.getState().getNostr().getPublicKey();
+  const hasPermission = !!findPermission?.enabled;
 
-    response.data = publicKey;
+  try {
+    if (hasPermission) {
+      const publicKey = state.getState().getNostr().getPublicKey();
+      return { data: publicKey };
+    } else {
+      const promptResponse = await utils.openPrompt<{
+        confirm: boolean;
+        rememberPermission: boolean;
+      }>({
+        args: {},
+        ...message,
+        action: "public/nostr/confirmGetPublicKey",
+      });
+      // add permission to db only if user decided to always allow this request
+      if (promptResponse.data.rememberPermission) {
+        const permissionIsAdded = await db.permissions.add({
+          createdAt: Date.now().toString(),
+          allowanceId: allowance.id,
+          host: message.origin.host,
+          method: PermissionMethod["NOSTR_GETPUBLICKEY"],
+          enabled: true,
+          blocked: false,
+        });
+
+        !!permissionIsAdded && (await db.saveToStorage());
+      }
 
-    return response;
+      if (promptResponse.data.confirm) {
```

Comment:
But currently this is always `true`, no?:  
https://github.com/getAlby/lightning-browser-extension/pull/1792/files#diff-d5402bf744dada20fceb38fbe951f38849c76376a409fbc8b3d32be73f58b724R28

Reply:
And if the prompt is cancelled this will never be reached?  
Or am I missing something?

Reply:
yes, I had a note somewhere. guess not in all screens.